libbeat/monitoring/report/report.go

```diff
@@ -25,6 +25,16 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
+// ReportingFormat encodes the type of format to report monitoring data in
+type ReportingFormat int
+
+// Enumerations of various ReportingFormats
+const (
+	ReportingFormatUnknown ReportingFormat = iota // to protect against zero-value errors
+	ReportingFormatXPackMonitoringBulk
+	ReportingFormatBulk
+)
```

Comment:
This one is very specific to the Elasticsearch reporter only. Can we hide this detail in the ES reporter plugin only?

nit: Always consider the package name to be part of the full name of a symbol. How about: `type EndpointType int`  enumeration with `UnknownEndpointType`, `XpackMonitoringEndpoint`, and `BulkEndpoint`?

Reply:
I tried to do push these constants down into the `monitoring/report/elasticsearch` package  but then I end up with import cycles, since these constants are used in the `monitoring/report` package, for e.g. [here](https://github.com/elastic/beats/pull/9260/files#diff-417858b4623eafaefe6fd1504259a6a3R99) and [here](https://github.com/elastic/beats/pull/9260/files#diff-417858b4623eafaefe6fd1504259a6a3R115).

This was why I kept these constants in the `monitoring/report` package and named them something more generic like "format" instead of "endpoint". The latter is definitely specific to Elasticsearch. By naming it "format", my thinking was that other monitoring reporters in the future could choose to handle the different reporting formats (or not).

Reply:
Well, alternative reporters are: write to log file, make available via HTTP REST API, and so on. No one besides the ES report is using any of these.
If we really need them to hack around limitation or how config is passed vs. to be interpreted, then it's fine for now. But in this case we should add a code comment explaining why it's here and if/how we plan to clean this up.

Reply:
As we discussed off-PR, I'm not happy about the hackiness of passing down this reporting "format", which is really only useful to the Elasticsearch reporter.

I couldn't come up with a really good solution without majorly rewriting how reporters configure themselves. However, I did a bit of cleanup and removed _some_ of the hackiness in 40e5163d976fe297ad00dc5f45a984d807b44125. Please let me know what you think of this solution.